src/components/GlobalPreferences/Network/Network.js

```diff
@@ -0,0 +1,164 @@
+import {
+  Box,
+  Button,
+  GU,
+  TextInput,
+  textStyle,
+  useLayout,
+  useTheme,
+} from '@aragon/ui'
+import { defaultEthNode, defaultIpfsGateway } from '../../../networks'
+import {
+  clearLocalStorageNetworkSettings,
+  setDefaultEthNode,
+  setIpfsGateway,
+} from '../../../local-settings'
+import { InvalidNetworkType, InvalidURI, NoConnection } from '../../../errors'
+import {
+  checkValidEthNode,
+  getNetworkType,
+  sanitizeNetworkType,
+} from '../../../lib/web3-utils'
+import { useEnterKey } from '../../../hooks/useKeyboardArrows'
+
+function Network() {
+  const {
+    ethNode,
+    networkType,
+    ipfsGateway,
+    handleEthNodeChange,
+    handleIpfsGatewayChange,
+    networkError,
+    handleNetworkChange,
+  } = useNetwork()
+  const theme = useTheme()
+
+  const { layoutName } = useLayout()
+  const compact = layoutName === 'small'
+
+  return (
+    <React.Fragment>
+      <Box heading="Node settings">
+        <Label theme={theme}>
+          Ethereum node
+          <TextInput
+            value={ethNode}
+            wide
+            onChange={handleEthNodeChange}
+            css={`
+              ${textStyle('body2')};
+              color: ${theme.contentSecondary};
+            `}
+          />
+          {networkError && (
+            <span
+              css={`
+                ${textStyle('body4')};
+                color: ${theme.negative};
+              `}
+            >
+              {(() => {
+                if (networkError instanceof InvalidNetworkType) {
+                  return `Node must be connected to ${sanitizeNetworkType(
+                    networkType
+                  )}`
+                }
+                if (networkError instanceof InvalidURI) {
+                  return 'Must provide Http endpoint to node'
+                }
+                if (networkError instanceof NoConnection) {
+                  return 'Could not connect to node'
+                }
+                return 'URI does not seem to be a ETH node'
+              })()}
+            </span>
+          )}
+        </Label>
+        <Label theme={theme}>
+          IPFS Gateway
+          <TextInput
+            value={ipfsGateway}
+            wide
+            onChange={handleIpfsGatewayChange}
+            css={`
+              ${textStyle('body2')};
+              color: ${theme.contentSecondary};
+            `}
+          />
+        </Label>
+        <Button mode="strong" onClick={handleNetworkChange} wide={compact}>
+          Save changes
+        </Button>
+      </Box>
+      <Box heading="Troubleshooting">
+        <div
+          css={`
+            margin-bottom: ${2 * GU}px;
+          `}
+        >
+          <span>
+            Press this button to reset the network settings to their defaults.
+          </span>
+        </div>
+        <Button
+          css={`
+            margin-bottom: ${2 * GU}px;
+          `}
+          onClick={clearLocalStorageNetworkSettings}
```

Comment:
Ahh, we should also refresh the app here! Otherwise this won't do anything until you do haha.